bin/package_build.py

```diff
@@ -49,82 +49,39 @@ def debian():
-	source_tumbleweed = [f"https://download.opensuse.org/ports/zsystems/tumbleweed/repo/oss/{x}" for x in ['s390x', 'noarch']]
-	source_leap_15_4 = [f"https://download.opensuse.org/distribution/leap/15.4/repo/oss/{x}" for x in ['s390x','noarch']]
-	for each in source_tumbleweed:
-		try:
-			req = requests.get(each)
-			data = req.text
-			if req.status_code == 404:
-				raise Exception("404 File not found")
-		except Exception as e:
-			print("Couldn't pull. Error: ",str(e))
-		else:
-			ref_data = re.findall(r"<a href=\"(.*\.rpm)\"><img.*<\/a>", data)
-			results.extend(ref_data)
-	DATA = open(file_name[0], 'w')
-	DATA.write('[')
-	for result in results:
-		result = re.sub(r'\.(noarch|s390x)\.rpm', '', result)
-		pkg = re.search(r'^([\w+\.-]+)-([+~\w\-\.]+)', result)
-		each_pkg = f'"packageName": "{pkg.group(1)}","version": "{pkg.group(2)}"'
-		each_pkg = '{'+each_pkg+'},'
-		DATA.write(each_pkg+'\n')
-	DATA.write('{}]')
-	DATA.close()
-	print(f"Saved!\nfilename: {q[0]}")
-	results = []
-	for each in source_leap:
-		try:
-			req = requests.get(each)
-			data = req.text
-			if req.status_code == 404:
-				raise Exception("404 File not found")
-		except Exception as e:
-			print("Couldn't pull. Error: ",str(e))
-		else:
-			ref_data = re.findall(r"<a href=\"(.*\.rpm)\"><img.*<\/a>", data)
-			results.extend(ref_data)
-	DATA = open(file_name[1], 'w')
-	DATA.write('[')
-	for result in results:
-		result = re.sub(r'\.(noarch|s390x)\.rpm', '', result)
-		pkg = re.search(r'^([\w+\.-]+)-([+~\w\-\.]+)', result)
-		each_pkg = f'"packageName": "{pkg.group(1)}","version": "{pkg.group(2)}"'
-		each_pkg = '{'+each_pkg+'},'
-		DATA.write(each_pkg+'\n')
-	DATA.write('{}]')
-	DATA.close()
-	print(f"Saved!\nfilename: {q[1]}")
-	results = []
-	for each in source_leap_15_4:
-		try:
-			req = requests.get(each)
-			data = req.text
-			if req.status_code == 404:
-				raise Exception("404 File not found")
-		except Exception as e:
-			print("Couldn't pull. Error: ",str(e))
-		else:
-			ref_data = re.findall(r"<a href=\"(.*\.rpm)\"><img.*<\/a>", data)
-			results.extend(ref_data)
-	DATA = open(file_name[2], 'w')
-	DATA.write('[')
-	for result in results:
-		result = re.sub(r'\.(noarch|s390x)\.rpm', '', result)
-		pkg = re.search(r'^([\w+\.-]+)-([+~\w\-\.]+)', result)
-		each_pkg = f'"packageName": "{pkg.group(1)}","version": "{pkg.group(2)}"'
-		each_pkg = '{'+each_pkg+'},'
-		DATA.write(each_pkg+'\n')
-	DATA.write('{}]')
-	DATA.close()
-	print(f"Saved!\nfilename: {q[2]}")
+	for i in range(len(source_data)):
+		opensuse_list=[]
+		for each in source_data[i]:
+			src_url =  each
+			try:
+				req = requests.get(src_url)
+				data_source = req.content
+				if req.status_code == 404:
+					raise Exception(f"404 Directory for Opensuse list not found")
+			except Exception as e:
+				print("Couldn't pull. Error: ",str(e))
+			else:
+				data = json.loads(data_source)
+				regex_pattern = r"-(.*?)-"
```

Comment:
@Princee215 could you move the `regex_pattern` above (its seems like a constant), ... define it just below `source_data` list